### PR TITLE
Complete `<h1>` settings heading

### PR DESCRIPTION
**Pull request by JuliaKirschenheuter**
* Resolves: https://github.com/nextcloud/server/issues/41826
* https://github.com/nextcloud/server/issues/41826#issuecomment-1888966567

## Summary

Before:

`<h1 class="hidden-visually" id="page-heading-level-1">Personal info</h1>`

After:

`<h1 class="hidden-visually" id="page-heading-level-1">Personal info: Personal settings</h1>`

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by JuliaKirschenheuter** [comment]
image

was approved by @jancborchardt

File: core/templates/layout.user.php [hunk 1]
```diff
@@ -87,7 +87,7 @@
 
 		<main id="content" class="app-<?php p($_['appid']) ?>">
 			<h1 class="hidden-visually" id="page-heading-level-1">
-				<?php p(!empty($_['pageTitle'])?$_['pageTitle']:$theme->getName()); ?>
+				<?php p(!empty($_['application'])?$_['application']:$theme->getName()); ?>: <?php p(!empty($_['pageTitle'])?$_['pageTitle']:$theme->getName()); ?>
```
[the remark this answers is not in the document]
Comment (ShGKme) [reply core/templates/layout.user.php:90]: Another problem could be with l10n. I'm not sure it is valid for any language to concatenate strings as
```
{title}: {getname}
```

**Review by jancborchardt — APPROVED** [review]

File: core/templates/layout.user.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (JuliaKirschenheuter) [reply core/templates/layout.user.php:90]: > Another problem could be with l10n. I'm not sure it is valid for any language to concatenate strings as

I've tested with German and seems to be fine

[the remark this answers is not in the document]
Comment (ShGKme) [reply core/templates/layout.user.php:90]: The issue with duplication seems to be quite a common problem. For example, the `<title>` currently is `Appearance and accessibility - Appearance and accessibility - Nextcloud` anyway.

The problem here is that `application` is defined by the active UserMenu entry...

But anyway, I think we should not write `{title}: {getname}` when they are the same to not have a `Dashboard: Dashboard` title by default.

**Review by artonge — APPROVED** [review]

File: core/templates/layout.user.php [hunk 2]
```diff
@@ -87,7 +87,10 @@
 
 		<main id="content" class="app-<?php p($_['appid']) ?>">
 			<h1 class="hidden-visually" id="page-heading-level-1">
-				<?php p(!empty($_['pageTitle'])?$_['pageTitle']:$theme->getName()); ?>
+				<?php p((!empty($_['application']) && !empty($_['pageTitle'] && $_['application'] != $_['pageTitle'])
+					? $_['application'].': '.$_['pageTitle']
+					: (!empty($_['pageTitle']) ? $_['pageTitle']
+					: $theme->getName())));	?>
```
Comment (ShGKme) [inline core/templates/layout.user.php:93]: A typo in brackets (second empty is not closed):
 
```suggestion
				<?php 
					p((!empty($_['application']) && !empty($_['pageTitle']) && $_['application'] != $_['pageTitle'])
						? $_['application'].': '.$_['pageTitle']
						: (!empty($_['pageTitle']) ? $_['pageTitle'] : $theme->getName())
					);
				?>
```

Comment (JuliaKirschenheuter) [reply core/templates/layout.user.php]: corrected, thanks a lot!

**Review by ShGKme — APPROVED** [review]

**Pull request merged**